src/components/orders/OrderDetails/OrderDetails.tsx

```diff
@@ -1,84 +1,95 @@
+import cn from 'classnames';
+import { Dispatch, FC, SetStateAction, useContext, useState } from 'react';
 
-import { orderService, userService } from '../../../api';
-import { IOrder } from '../../../types';
+import { CurrentUserContext } from '../../../context';
+import { IOrder, IPaymentPayload, PaymentCurrency } from '../../../types';
 import { ProgressBar } from '../../common';
+import { Payment } from '../../payment/Payment';
 import UpdateOrder from '../UpdateOrder/UpdateOrder';
 import style from './OrderDetails.module.scss';
 
 
 interface IOrderCardProps {
   order: IOrder;
   setOrder: Dispatch<SetStateAction<IOrder>>;
-  id: number;
 }
 
-const OrderDetails: FC<IOrderCardProps> = ({ order, setOrder, id }) => {
+const OrderDetails: FC<IOrderCardProps> = ({ order, setOrder }) => {
+  const { user } = useContext(CurrentUserContext)
+  const [open, setOpen] = useState(false);
+  const [payment, setPayment] = useState(false);
+  const [amount, setAmount] = useState(0);
+  const [currency, setCurrency] = useState<PaymentCurrency>(PaymentCurrency.UAH);
+  const [paymentPayload, setPaymentPayload] = useState<IPaymentPayload>()
 
-  const [orderDb, setOrderDb] = useState<boolean>(false);
-  const [open, setOpen] = useState<boolean>(false);
-  const navigate = useNavigate();
-
-  const getUser = async () => {
-    const userFromDb = await userService.retrieve(localStorage.getItem('token'));
-    if (userFromDb.role === 'volunteer') {
-      const orderFromDB = await orderService.getUserOrder(id);
-      if (orderFromDB) {
-        setOrderDb(true);
-      }
+  const submitPayment = () => {
+    if (amount && order?.title && order?.id) {
+      setPaymentPayload(
+        { amount, currency, description: order?.title, order_id: order?.id })
+      setPayment(true)
     }
-  };
+  }
 
-  useEffect(() => {
-    if (localStorage.getItem('token')) {
-      getUser();
-    } else {
-      navigate('/sign-in');
-    }
-  }, []);
-
-  const handleOpen = () => {
-    setOpen(true);
-  };
-  const handleClose = () => {
-    setOpen(false);
-  };
+  const handleClose = () => setOpen(false);
 
   return (
     <div className={style.container}>
-      <div className={style.title}>
-        <h1>{order?.title}</h1>
-        <p>{order?.info}</p>
-      </div>
-      <div className={style.main}>
-        <div className={style.image}
-          style={{ backgroundImage: `url(${order?.photo})` }}>
-        </div>
-        <div className={style.content}>
-          <div className={style.content_header}>
-            <div className=
-              {order?.status === 'open' ?
-                style.gathering_open : style.gathering_close}>
-              <h4>{order?.status === 'open' ? 'Відкритий збір' : 'Збір закрито'}</h4>
+      {
+        payment ?
+          <Payment paymentPayload={paymentPayload} /> :
+          <div>
+            <ProgressBar moneyHave={order?.sum} moneyNeed={order?.goal_amount}
```

Comment:
```suggestion
            <ProgressBar
                moneyHave={order?.sum}
                moneyNeed={order?.goal_amount}
                closedAt={new Date(order?.finished_at)}
                size='large'
            />
```
will be more easy to use